src/Compilers/Core/MSBuildTaskTests/DotNetSdkTests.cs

```diff
@@ -0,0 +1,288 @@
+                    $@"{root1}sub1\: /_/sub1/",
+                    $@"{root1}sub2\: /_/sub2/",
+                    @"true",
+                    $@"{root2}=/_1/,{root1}=/_/"
+                });
+
+            AssertEx.AssertEqualToleratingWhitespaceDifferences(
+                "[/_1/]=[https://raw.githubusercontent.com/Source/Package/*]," +
+                "[/_/]=[https://raw.githubusercontent.com/R1/*]," +
+                "[/_/sub1/]=[https://raw.githubusercontent.com/M1/*]," +
+                "[/_/sub2/]=[https://raw.githubusercontent.com/M2/*]",
+                File.ReadAllText(sourceLinkJsonPath));
+
+            // non-deterministic CI build:
+            VerifyValues(
+                props: $@"
+<Project>
+  <PropertyGroup>
+    <SourceControlInformationFeatureSupported>true</SourceControlInformationFeatureSupported>
+    <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
+    <Deterministic>false</Deterministic>
+  </PropertyGroup>
+  {sourcePackageTargets}
+  {sourceLinkPackageTargets}
+</Project>",
+                targets: new[]
+                {
+                    "CoreCompile"
+                },
+                expressions: new[]
+                {
+                    "@(SourceRoot->'%(Identity): %(MappedPath)')",
+                    "$(DeterministicSourcePaths)",
+                    "$(PathMap)"
+                },
+                expectedResults: new[]
+                {
+                    $@"{root2}: {root2}",
+                    $@"{root1}: {root1}",
+                    $@"{root1}sub1\: {root1}sub1\",
+                    $@"{root1}sub2\: {root1}sub2\",
+                    @"",
+                    $@""
+                });
+
+            AssertEx.AssertEqualToleratingWhitespaceDifferences(
+                $@"[{root2}]=[https://raw.githubusercontent.com/Source/Package/*]," +
+                $@"[{root1}]=[https://raw.githubusercontent.com/R1/*]," +
+                $@"[{root1}sub1\]=[https://raw.githubusercontent.com/M1/*]," +
+                $@"[{root1}sub2\]=[https://raw.githubusercontent.com/M2/*]",
+                File.ReadAllText(sourceLinkJsonPath));
+
+            // deterministic local build:
+            VerifyValues(
+                props: $@"
+<Project>
+  <PropertyGroup>
+    <SourceControlInformationFeatureSupported>true</SourceControlInformationFeatureSupported>
+    <ContinuousIntegrationBuild>false</ContinuousIntegrationBuild>
+  </PropertyGroup>
+  {sourcePackageTargets}
+  {sourceLinkPackageTargets}
+</Project>",
+                targets: new[]
+                {
+                    "CoreCompile"
+                },
+                expressions: new[]
+                {
+                    "@(SourceRoot->'%(Identity): %(MappedPath)')",
+                    "$(DeterministicSourcePaths)",
+                    "$(PathMap)"
+                },
+                expectedResults: new[]
+                {
+                    $@"{root2}: {root2}",
```

Comment:
Nit: Perhaps remove the `@` symbols on single-line string literals

Reply:
They are useful for avoiding escaping of `\` in the other cases.